databases/week3/sql_week3.sql

```diff
@@ -0,0 +1,237 @@
+('Vegetable Stew', ' Vegetable Stew is a combination of vegetable creamy curry prepared with thick and thin coconut milk and dry spices and served with rice', 'Valby', '2021-07-16 18:30:00', 12, 120, now()),
+('Chicken Kadai ', 'Kadai chicken is a delicious, spicy & flavorful dish made with chicken, onions, tomatoes, ginger, garlic & fresh ground spices known as kadai masala.', 'Frederiksberg', '2021-07-14 18:00:00',12 , 110, now());
+
+
+INSERT INTO reservation(number_of_guests, created_date, contact_phonenumber, contact_name, contact_email, meal_id)
+ VALUES (5, now(), '91352857','Dumbell Shiro', 'dshiro@spotify.com', 15),
+ (4, now(), '89152968','Guruwain Kumar', 'gKumar@phoca.cz', 16),
+ (3, now(), '67456795','Kyle Bessler',  'kbessler@chronoengine.com',17),
+ (2, now(), '34329630','Turdy Ray',  'tRay@harvard.edu', 18);
+ 
+INSERT INTO reservation(number_of_guests, created_date, contact_phonenumber, contact_name, contact_email, meal_id)
+ VALUES (9, now(), '91358597','Biju Sambhar', 'bsambhar@spotify.com', 17);
+ 
+
+
+ INSERT INTO review(title, description, meal_id, stars)
+ VALUES ('Excellent food', 'What a lovely homely expereince. It was food for my soul ', 15, 5), 
+ ('Will do it again', 'That was some lovley food: i recommend everone to try it out ', 16, 4), 
+ ('Yummy food', 'It was good, but not the best ', 17, 3),
+  ('It was average', 'It was average. The food had no flavour ', 17, 2);
+  
+-- Get meals that has a price smaller than a specific price fx 90
+SELECT meal.title, meal.price
+FROM meal
+WHERE price<90;
+
+-- Get meals that still has available reservations
+SELECT meal.id, meal.title, meal.max_reservations AS max_guests, SUM(reservation.number_of_guests) AS total_reserved
+FROM meal
+INNER JOIN reservation ON meal.id = reservation.meal_id
+GROUP BY meal.title
+HAVING total_reserved < max_guests;
+
+-- Get meals that partially match a title. Rød grød med will match the meal with the title Rød grød med fløde
+
+SELECT title
+FROM meal
+WHERE title LIKE "Chicken%";
+
+-- Get only specific number of meals fx return only 5 meals
+SELECT title, description
+FROM meal 
+LIMIT 5;
+
+-- Get meals that has been created between two dates
+SELECT meal.title, DAY(reservation.created_date) AS created_on
+FROM meal
+INNER JOIN reservation ON meal.id = reservation.meal_id
+HAVING created_on > 14 && created_on <20;
+
+-- Get the meals that have good reviews
+SELECT meal.title, review.stars
+FROM meal
+INNER JOIN review ON meal.id = review.meal_id
+WHERE stars > 4;
+
+-- Get reservations for a specific meal sorted by created_date
+SELECT meal.title AS name, reservation.created_date AS created_on
+FROM meal 
+INNER JOIN reservation ON meal.id = reservation.meal_id
+ORDER BY created_on DESC;
+
+-- Sort all meals by average number of stars in the reviews
+SELECT meal.title, AVG(review.stars)
+FROM meal 
+INNER JOIN review ON meal.id = review.meal_id
+GROUP BY meal.title
+ORDER BY review.stars DESC;
```

Comment:
You should also consider the meals without reviews in this query. How would you achieve that?